### PR TITLE
Split meteor in three parts

### PR DESCRIPTION
**Pull request by jfsiebel**
This PR split the meteor into three different parts, each part have a specific role:

**Backend**: This meteor is responsible for handling the `redis` messages from `akka-apps` and updating mongo.

**Whiteboard**: This meteor handle all whiteboard and cursor drawing/move events.

**Frontend**: Interact with **Backend** and **Whiteboard**, watch mongo changes in order to update the informations for the user.

### Why?
Current we have the same node process to handle everything in `bbb-html5`, because of that all messages go to the same queue, causing some messages to have a delay until handled.

### How
With the implementation of different meteors we are able to handle more messages, because the `whiteboard/cursor` messages are handled in a different queue and process than other messages.

### Testing
In order to test this PR it'll be needed to add a rule on nginx(`/etc/bigbluebutton/nginx/bbb-html5.nginx`):
```
location /html5client-wb {
  proxy_pass http://127.0.0.1:3200;
  proxy_http_version 1.1;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection "Upgrade";
}
```

To run meteor from the source you'll need to run `npm run start:all`

In order to run each meteor instance it'll be needed some environment variables:
- Backend:
  - METEOR_ROLE=backend
  - PORT=3100
- Whiteboard
   - METEOR_ROLE=whiteboard
  - PORT=3200
- Frontend
   - METEOR_ROLE=frontend

### CODE REVIEW

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/core/util/jhotdraw/Bezier.java [hunk 1]
```diff
@@ -21,21 +21,6 @@
...
- * of Hand-Sketched Curves.<br>
- * © Copyright by Philip J. Schneider 1988.<br>
- * A thesis submitted in partial fulfillment of the requirements for the degree
- * of Master of Science, University of Washington.
- * <p>
- * http://autotrace.sourceforge.net/Interactive_Curve_Design.ps.gz
- *
- * @author Werner Randelshofer
- * @version $Id$
- */
```
Comment (capilkey) [inline akka-bbb-apps/src/main/scala/org/bigbluebutton/core/util/jhotdraw/Bezier.java]: I'm going to guess that these blocks were auto-removed by some script on your computer. Could please add them back

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/core/util/jhotdraw/Bezier.java [hunk 2]
```diff
@@ -229,19 +214,6 @@ public static BezierPath fitBezierPath(BezierPath digitizedPoints, double error)
...
-     * Corner points are both contained as the last point of a segment and
-     * the first point of a subsequent segment.
-     * 
-     * @param digitizedPoints Digitized points 
-     * @param maxAngle maximal angle in radians between the current point and its
-     * predecessor and successor up to which the point does not break the
-     * digitized list into segments. Recommended value 44° = 44 * 180d / Math.PI
-     * @return Segments of digitized points, each segment having less than maximal
-     * angle between points.
-     */
```
Comment (capilkey) [inline akka-bbb-apps/src/main/scala/org/bigbluebutton/core/util/jhotdraw/Bezier.java]: Add this one back also please

**Comment by jibon57** [comment]
Excellent job @jfsiebel . Is there ETA to merge this? So, now BBB will use 3 separate processes to achieve multiple threads? Am I understanding correctly?

**Comment by amguirado73** [comment]
Hello,

I have studied deeply this PR and create a bbb version with this changes but it doesn't work. The main issue is that fronted process is not subscribe to redis channels. Frontend process needs some messages (ValidateAuthToken) coming from akka-apps because it has the pendingAuthTokens and must join the user to meeting. When a user go into BBB will have a screen with dots and nothing more.

Regards

**Comment by pedrobmarin** [comment]
https://github.com/bigbluebutton/bigbluebutton/pull/11317 seem to have captured the idea of this. @jfsiebel may we close this?

**Comment by jfsiebel** [comment]
I think so.

**Pull request closed without merging**